plugins/performance-lab/includes/site-health/far-future-headers/helper.php

```diff
@@ -0,0 +1,302 @@
+				esc_html__( 'Note: "Conditionally cached" means that the browser can re-validate the resource using ETag or Last-Modified headers. This results in fewer full downloads but still requires the browser to make requests, unlike far-future expiration headers that allow the browser to fully rely on its local cache for a longer duration.', 'performance-lab' )
+			);
+		} else {
+			$result['actions'] = sprintf(
+				'<p>%s</p>',
+				esc_html__( 'Far-future Cache-Control or Expires headers can be added or adjusted with a small configuration change by your hosting provider.', 'performance-lab' )
+			);
+		}
+	}
+
+	return $result;
+}
+
+/**
+ * Checks if far-future expiration headers are enabled for a list of assets.
+ *
+ * @since n.e.x.t
+ *
+ * @param  string[] $assets List of asset URLs to check.
+ * @return array{final_status: string, details: array{filename: string, reason: string}[]} Final status and details.
+ */
+function perflab_ffh_check_assets( array $assets ): array {
+	$final_status = 'good';
+	$fail_details = array(); // Array of arrays with 'filename' and 'reason'.
+
+	foreach ( $assets as $asset ) {
+		$response = wp_remote_get( $asset, array( 'sslverify' => false ) );
+
+		// Extract filename from the URL.
+		$path_info = pathinfo( (string) wp_parse_url( $asset, PHP_URL_PATH ) );
+		$filename  = isset( $path_info['basename'] ) ? $path_info['basename'] : basename( $asset );
+
+		if ( is_wp_error( $response ) ) {
+			// Can't determine headers if request failed, consider it a fail.
+			$final_status   = 'recommended';
+			$fail_details[] = array(
+				'filename' => $filename,
+				'reason'   => __( 'Could not retrieve headers', 'performance-lab' ),
+			);
+			continue;
+		}
+
+		$headers = wp_remote_retrieve_headers( $response );
+		if ( ! is_object( $headers ) && 0 === count( $headers ) ) {
+			// No valid headers retrieved.
+			$final_status   = 'recommended';
+			$fail_details[] = array(
+				'filename' => $filename,
+				'reason'   => __( 'No valid headers retrieved', 'performance-lab' ),
+			);
+			continue;
+		}
+
+		$check = perflab_ffh_check_headers( $headers );
+		if ( isset( $check['passed'] ) && $check['passed'] ) {
+			// This asset passed far-future headers test, no action needed.
+			continue;
+		}
+
+		// If not passed, decide whether to try conditional request.
+		if ( false === $check ) {
+			// Only if no far-future headers at all, we try conditional request.
+			$conditional_pass = perflab_ffh_try_conditional_request( $asset, $headers );
+			if ( ! $conditional_pass ) {
+				$final_status   = 'recommended';
+				$fail_details[] = array(
+					'filename' => $filename,
+					'reason'   => __( 'No far-future headers and no conditional caching', 'performance-lab' ),
+				);
+			} else {
+				$final_status   = 'recommended';
+				$fail_details[] = array(
+					'filename' => $filename,
+					'reason'   => __( 'No far-future headers but conditionally cached', 'performance-lab' ),
+				);
+			}
```

Comment:
Factoring `$final_status` out of the `if` statement.

```suggestion
			$conditional_pass = perflab_ffh_try_conditional_request( $asset, $headers );
			$final_status     = 'recommended';
			if ( ! $conditional_pass ) {
				$fail_details[] = array(
					'filename' => $filename,
					'reason'   => __( 'No far-future headers and no conditional caching', 'performance-lab' ),
				);
			} else {
				$fail_details[] = array(
					'filename' => $filename,
					'reason'   => __( 'No far-future headers but conditionally cached', 'performance-lab' ),
				);
			}
```